test/common/testlib.py

```diff
@@ -476,16 +476,25 @@ def logout(self):
         if self.is_visible("button#machine-reconnect"):
             # happens when shutting down cockpit or rebooting machine
             self.click("button#machine-reconnect")
+            self.expect_load()
+            return False
         else:
             # happens when cockpit is still running
             self.click("#navbar-dropdown")
             self.click('#go-logout')
-        self.expect_load()
+            self.expect_load()
+            self.expect_load()
```

Comment:
This illustrates the problem, and it might even be a race condition? If client.Page.loadEventFired() gets called twice in quick succession while the second pageLoadResolve promise isn't yet set up? Maybe it "usually works", but it's still weird that the page loads itself twice.